rum-injection-installer/installer/nginx_configurator.go

```diff
@@ -0,0 +1,272 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+type NginxConfigurator struct {
+	Version     string
+	ModulesPath string
+}
+
+// Runs 'nginx -v' and looks for the nginx version.
+// Checks that the version is higher than a min version
+func (n *NginxConfigurator) Verify() error {
+	cmd := exec.Command("nginx", "-v")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("nginx is not installed")
+	}
+
+	versionRegex := regexp.MustCompile(`nginx version: nginx/(\d+\.\d+\.\d+)`)
+	matches := versionRegex.FindStringSubmatch(string(output))
+	if len(matches) < 2 {
+		return fmt.Errorf("failed to parse nginx version: %s", output)
+	}
+
+	n.Version = matches[1]
+
+	print("Detected NGINX version: ", n.Version, "\n")
+
+	const minSupportedVersion = "1.22.0"
+	if compareVersions(n.Version, minSupportedVersion) < 0 {
+		return fmt.Errorf("nginx version %s is not supported, must be at least %s", n.Version, minSupportedVersion)
+	}
+
+	return nil
+}
+
+// Reads the config file
+// 1. Checks if a line containing the module already exists
+// 2. Creates a backup of the config file
+// 3. Appends the load_module line to the beginning of the file
+// 4. Inserts the RUM config as early as possible in the http section
+func (n *NginxConfigurator) ModifyConfig(appID, site, clientToken, agentUrl string, sessionSampleRate, sessionReplaySampleRate int) error {
+	configPath, err := n.getConfigLocation()
+	if err != nil {
+		return fmt.Errorf("failed to get config location: %v", err)
+	}
+
+	content, err := os.ReadFile(configPath)
+	if err != nil {
+		return fmt.Errorf("failed to read config file: %v", err)
+	}
+
+	// Check if Datadog module is already loaded
+	if strings.Contains(string(content), "ngx_http_datadog_module.so") {
+		return fmt.Errorf("datadog module is already loaded in the configuration")
+	}
```

Comment:
The following may return a false positive if the Datadog module is named differently. To ensure the module isn't already loaded, you can attempt to load it—NGINX will return an error if it is already installed.

```sh
# the following will fail if the default config already load the datadog module
nginx -g "load_module <MODULE>;" -t

nginx: [emerg] module "ngx_http_datadog_module" is already loaded in /Users/damien.mehala/workspace/nginx-datadog/nginx.conf:2
```